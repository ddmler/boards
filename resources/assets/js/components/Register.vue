<template>
    <div>
        <div v-if="error && !success" class="notification is-danger">
            <p>There was an error. Please check the error messages below.</p>
        </div>
        <div v-if="success" class="notification is-success">
            <p>Registration successful. You can now <router-link :to="{name:'login'}">login.</router-link></p>
        </div>
        <form autocomplete="off" @submit.prevent="register" v-if="!success" method="post">
            <div class="field">
                <label for="name" class="label">Username</label>
                <input type="text" id="name" v-model="name" required :class="[{ 'is-danger': error && errors.name }, 'input']">
                <div v-if="error && errors.name" class="message is-danger is-small"><div class="message-body">{{ errors.name }}</div></div>
            </div>
            <div class="field">
                <label for="email" class="label">E-mail</label>
                <input type="email" id="email" placeholder="user@example.com" v-model="email" required :class="[{ 'is-danger': error && errors.email }, 'input']">
                <div v-if="error && errors.email" class="message is-danger is-small"><div class="message-body">{{ errors.email }}</div></div>
            </div>
            <div class="field">
                <label for="password" class="label">Password</label>
                <input type="password" id="password" v-model="password" required :class="[{ 'is-danger': error && errors.password }, 'input']">
                <div v-if="error && errors.password" class="message is-danger is-small"><div class="message-body">{{ errors.password }}</div></div>
            </div>
            <button type="submit" class="button">Register</button>
        </form>
    </div>
</template>
<script>
    export default {
        data(){
            return {
                name: '',
                email: '',
                password: '',
                error: false,
                errors: {},
                success: false
            };
        },
        methods: {
            register(){
                var app = this
                this.$auth.register({
                    params: {
                        name: app.name,
                        email: app.email,
                        password: app.password
                    },
                    success: function () {
                        app.success = true
                    },
                    error: function (resp) {
                        app.error = true;
                        app.errors = resp.response.data.errors;
                    },
                    redirect: null
                });
            }
        }
    }
</script>
