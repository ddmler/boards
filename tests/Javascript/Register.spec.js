import { mount } from '@vue/test-utils';
import Register from '@/Register.vue';

describe('Register.vue', () => {

    it('matches the snapshot', () => {
        const wrapper = mount(Register)
        expect(wrapper.html()).toMatchSnapshot();
    });

});