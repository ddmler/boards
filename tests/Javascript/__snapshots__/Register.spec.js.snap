// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue matches the snapshot 1`] = `
<div>
  <!---->
  <form autocomplete="off" method="post">
    <div class="field"><label for="name" class="label">Username</label> <input id="name" type="text" required="required" class="input"></div>
    <div class="field"><label for="email" class="label">E-mail</label> <input id="email" type="email" placeholder="user@example.com" required="required" class="input"></div>
    <div class="field"><label for="password" class="label">Password</label> <input id="password" type="password" required="required" class="input"></div> <button type="submit" class="button">Register</button>
  </form>
</div>
`;
